script/testing/artifact_stats/common_collector_functions.py

```diff
@@ -0,0 +1,28 @@
+import subprocess
+from util.constants import LOG
+
+
+def compile_binary(build_path, use_cache=True, is_debug=False):
+    """ Compile binary 
+    Args:
+        build_path - The location of the binary
+        use_cache - Whether to increase the speed by using the build cache
+        is_debug - Whether to output the logs of the compilation to stdout
+    Retrun:
+        int - The exit code of the compilation (0=success)
+    """
+    cmake_cmd = f'cmake -GNinja -DNOISEPAGE_UNITY_BUILD=ON \
+        {"-DCMAKE_CXX_COMPILER_LAUNCHER=ccache" if use_cache else ""} -DCMAKE_BUILD_TYPE=Release \
+        -DNOISEPAGE_USE_ASAN=OFF -DNOISEPAGE_USE_JEMALLOC=ON ..'
+    ninja_cmd = 'ninja noisepage'
+    compile_commands = [cmake_cmd, ninja_cmd]
```

Comment:
Yah I think there are a couple of options.
1) I could pass a string as a command-line arg to the script which contains the build commands
       - Its not the most elegant solution and we might have to pass multiple args to be able to turn on and off caching
2) I could create a build script
       - This way if we wanted to make changes to the build commands there is only one place to make those changes. 
          The normal build, nightly build, and these tests could use the build script.
       - You still could have the problem where if there are drastic changes to the arguments then these tests might not 
          be updated. 
3) We could set it up so the nightly build will send a slack message when it fails so people are aware if things like this get broken. 
       - I could do something where I read through the Jenkinsfile to see if the build command matches a build command 
          in the Jenkinsfile and throw an error if it doesn't. 
4) I could do all the metric collection directly in the Jenkinsfile-nightly.
       - This would make the file pretty unwieldy and much harder for people to test before committing changes

I can keep thinking about if there are better solutions.

Reply:
I could also store the build command in an environment variable in the Jenkinsfile and use that in the script